components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/PostAuthnMissingClaimHandler.java

```diff
@@ -288,6 +311,15 @@ protected void handlePostAuthenticationForMissingClaimsResponse(HttpServletReque
 
         if (!doMandatoryClaimsExist) {
             // Check whether mandatory claims exist in the request. If not throw error.
+            if (LoggerUtils.isDiagnosticLogsEnabled()) {
+                Map<String, Object> params = new HashMap<>();
+                params.put(FrameworkConstants.LogConstants.SERVICE_PROVIDER, context.getServiceProviderName());
+                params.put(FrameworkConstants.LogConstants.TENANT_DOMAIN, context.getTenantDomain());
+                LoggerUtils.triggerDiagnosticLogEvent(
+                        FrameworkConstants.LogConstants.AUTHENTICATION_FRAMEWORK, params, LogConstants.FAILED,
+                        "Mandatory missing claims are not found",
```

Comment:
Same here.. can we say which are missing?

Reply:
In this case, we are checking the response params to see if we have the mandatory params included there. These are sent when the user is prompted to enter the mandatory params. So, the handler knows it must return at least one mandatory claim in the response. The way it sends the mandatory claims is by prepending the prefix : `claim_mand~`
If the response does not contain any params, the handler will throw an error and break the flow.
Therefore, the missing params are nort available in the context here